src/main/java/com/eerussianguy/blazemap/mixin/ServerLevelMixin.java

```diff
@@ -1,26 +1,44 @@
 package com.eerussianguy.blazemap.mixin;
 
+import java.util.List;
+import java.util.concurrent.Executor;
+
 import net.minecraft.core.BlockPos;
+import net.minecraft.core.Holder;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.progress.ChunkProgressListener;
+import net.minecraft.world.level.Level;
 import net.minecraft.world.level.ChunkPos;
+import net.minecraft.world.level.CustomSpawner;
 import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.chunk.ChunkGenerator;
+import net.minecraft.world.level.dimension.DimensionType;
+import net.minecraft.world.level.storage.LevelStorageSource;
+import net.minecraft.world.level.storage.ServerLevelData;
 
 import com.eerussianguy.blazemap.engine.server.BlazeMapServerEngine;
+
 import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 @Mixin(ServerLevel.class)
-public abstract class ServerLevelMixin {
+public abstract class ServerLevelMixin extends Level {
+    // This won't be called because it's a mixin, but Java still wants it to exist at compile time,
+    // so directly copying from ServerLevel
+    public ServerLevelMixin(MinecraftServer p_203762_, Executor p_203763_, LevelStorageSource.LevelStorageAccess p_203764_, ServerLevelData p_203765_, ResourceKey<Level> p_203766_, Holder<DimensionType> p_203767_, ChunkProgressListener p_203768_, ChunkGenerator p_203769_, boolean p_203770_, long p_203771_, List<CustomSpawner> p_203772_, boolean p_203773_) {
+      super(p_203765_, p_203766_, p_203767_, p_203762_::getProfiler, false, p_203770_, p_203771_);
```

Comment:
Could be a default constructor (no args) that just passes null / false / 0 everywhere...
Since it's not used, it's just meant to shut up the compiler.
That'd also make us not have to import 15 classes we're not actually using.
Functionally it makes absolutely no difference whatsoever, but I'd rather keep the file as simple as possible so it's easy to focus on what matters.

Reply:
It can't be empty as Java won't let it compile without a call to `super()`, and `Level` (the superclass) requires args in its constructor. Also, as this class doesn't exist at runtime, it doesn't matter that all these extra imports are used here because the imports are happening anyway in `ServerLevel`. 

That being said, now I'm no longer in "throw everything at the wall to make it work" mode, I can get it so it takes in only the args it needs to directly pass along to `Level`'s constructor. And I might also test if I can get away with _not_ extending `Level` to remove that complexity further, but I'd like to do things the way the docs say to where possible, even if it's technically possible to do without them, because the docs represent the "expected behaviour" when interacting with a system.

Reply:
I meant like this
```suggestion
    public ServerLevelMixin() {
        super(null, null, null, null, false, null, null);
    }
```
Even if the compiler bitches and moans about all the nulls in places that probably aren't nullable 😛

Reply:
Ah gotcha! Yep, can do that